bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusVMB4ANHandler.java

```diff
@@ -0,0 +1,271 @@
+                    }
+                }
+
+                channelText[channel - 9] = channelText[channel - 9].substring(0, textStartPosition)
+                        + contents.toString()
+                        + (channelText[channel - 9].length() > textStartPosition + 5 ? channelText[channel - 9]
+                                .substring(textStartPosition + 5, channelText[channel - 9].length()) : "");
+
+                String channelUID = convertAnalogInputChannelByteToChannelUID(channel);
+                updateState(channelUID, new StringType(channelText[channel - 9]));
+            }
+        }
+    }
+
+    protected byte convertChannelUIDToChannelByte(ChannelUID channelUID) {
+        if (isAlarmChannel(channelUID)) {
+            return convertAlarmChannelUIDToChannelByte(channelUID);
+        } else if (isTextAnalogInputChannel(channelUID)) {
+            return convertTextAnalogInputChannelUIDToChannelByte(channelUID);
+        } else if (isRawAnalogInputChannel(channelUID)) {
+            return convertRawAnalogInputChannelUIDToChannelByte(channelUID);
+        } else if (isAnalogOutputChannel(channelUID)) {
+            return convertAnalogOutputChannelUIDToChannelByte(channelUID);
+        } else {
+            throw new UnsupportedOperationException(
+                    "The channel '" + channelUID + "' is not supported on a VMB4AN module.");
+        }
+    }
+
+    protected boolean isAlarmChannel(ChannelUID channelUID) {
+        return channelUID.getGroupId() == ALARM_GROUP;
+    }
+
+    protected byte convertAlarmChannelUIDToChannelByte(ChannelUID channelUID) {
+        return Byte.parseByte(channelUID.getIdWithoutGroup());
+    }
+
+    protected boolean isTextAnalogInputChannel(ChannelUID channelUID) {
+        return channelUID.getGroupId() == ANALOG_INPUT_GROUP
+                && !channelUID.getIdWithoutGroup().endsWith(RAW_CHANNEL_SUFFIX);
+    }
+
+    protected boolean isRawAnalogInputChannel(ChannelUID channelUID) {
+        return channelUID.getGroupId() == ANALOG_INPUT_GROUP
+                && channelUID.getIdWithoutGroup().endsWith(RAW_CHANNEL_SUFFIX);
+    }
+
+    protected byte convertRawAnalogInputChannelUIDToChannelByte(ChannelUID channelUID) {
+        return Byte.parseByte(channelUID.getIdWithoutGroup().replaceAll(RAW_CHANNEL_SUFFIX, ""));
+    }
+
+    protected byte convertTextAnalogInputChannelUIDToChannelByte(ChannelUID channelUID) {
+        return Byte.parseByte(channelUID.getIdWithoutGroup());
+    }
+
+    protected String convertAnalogInputChannelByteToRawChannelUID(byte channelByte) {
+        return convertAnalogInputChannelByteToChannelUID(channelByte) + RAW_CHANNEL_SUFFIX;
+    }
+
+    protected String convertAnalogInputChannelByteToChannelUID(byte channelByte) {
+        return ANALOG_INPUT_GROUP + "#" + channelByte;
+    }
+
+    protected boolean isAnalogOutputChannel(ChannelUID channelUID) {
+        return channelUID.getGroupId() == ANALOG_OUTPUT_GROUP;
```

Comment:
String comparison should not be using `==`:
```suggestion
        return ANALOG_OUTPUT_GROUP.equals(channelUID.getGroupId());
```

Reply:
Modified in commit d57ba9d